ee/src/main/java/org/jboss/as/ee/concurrent/ElytronManagedThreadFactory.java

```diff
@@ -41,51 +40,25 @@
  */
 public class ElytronManagedThreadFactory extends ManagedThreadFactoryImpl {
 
-    private static final ClassLoader CLASSLOADER = ElytronManagedThreadFactory.class.getClassLoader();
-
     public ElytronManagedThreadFactory(String name, ContextServiceImpl contextService, int priority) {
         super(name, contextService, priority);
     }
 
-    protected AbstractManagedThread createThread(final Runnable r, final ContextHandle contextHandleForSetup) {
-        boolean checking = WildFlySecurityManager.isChecking();
-        SecurityDomain domain = checking ?
-                AccessController.doPrivileged((PrivilegedAction<SecurityDomain>) SecurityDomain::getCurrent) :
-                SecurityDomain.getCurrent();
-        SecurityIdentity identity = domain == null ? null : domain.getCurrentSecurityIdentity();
-        final ClassLoader tccl = WildFlySecurityManager.getCurrentContextClassLoaderPrivileged();
-        try {
-            WildFlySecurityManager.setCurrentContextClassLoaderPrivileged(CLASSLOADER);
-            if (checking) {
-                return AccessController.doPrivileged((PrivilegedAction<ElytronManagedThread>)
-                        () -> new ElytronManagedThread(r, contextHandleForSetup, identity)
-                );
-            } else {
-                return new ElytronManagedThread(r, contextHandleForSetup, identity);
-            }
-        } finally {
-            WildFlySecurityManager.setCurrentContextClassLoaderPrivileged(tccl);
+    protected AbstractManagedThread createThread(Runnable r, final ContextHandle contextHandleForSetup) {
+        if (contextHandleForSetup != null) {
+            // app thread, do identity wrap
+            r = SecurityIdentityUtils.doIdentityWrap(r);
```

Comment:
This doesn't look right to me. The `Runnable` will already be wrapped in the `ManagedExecutorServiceImpl`. I don't think we need it here as it seems it could lead to the same identity issue we already see.

Reply:
If handle is not null it’s not a new thread request from executor, it’s a new thread request from app, which may obtain and use the threadfactory directly.

Reply:
Do we know for sure that the wrapped runnable itself will exit after it's complete? I'd have to dig heavily through the code to find out. I'm only looking for confirmation that we know that `Runnable` passed in will exit :)

Reply:
We decided long ago to not make such control, apps are responsible for their tasks. Anyway don't think that's relevant for the issue, threads created directly by apps are not shared, and will run with this identity while alive, while threads created by executors and shared by apps won't have an initial identity set (which should solve any related leaks too), each task submitted by apps to executor will have their own identity wrap done at executor.